module-2/backend/week_four.md

```diff
@@ -10,14 +10,40 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+Data that's generated by a web site and stored on the machine to identify them whenever they visit
+
 * What’s the difference between a session and a cookie?
+A session is a cookie that's been hashed
+
 * What’s a flash and when do you want to use flashes?
+A message that can be displayed after an action is taken. I'd use flashes to tell a user whether the action they tried to complete was successful or not. Also, I'd like to warn them whenever they're trying to do something that will change the database (updating or deleting).
+
 * Why do people say “HTTP is stateless”?
+HTTP is called stateless because we don't have to keep track of a session for our users to view pages on our site
```

Comment:
each individual request/response cycle is independent of the next. it doesn't track any data between requests so we need sessions and cookies to help us maintain state